refactor(language_server): handle `ServerLinter` it own restart on `workspace/didChangeConfiguration`

Sysix opened the pull request:
> This PR refactors the configuration change handling logic by extracting the linter restart logic into a new `handle_configuration_change` method and introducing a `ToolRestartChanges` struct to encapsulate the changes.

The struct will later be used for the Formatter too.

camc314 commented:
### Merge activity

* **Nov 7, 2:56 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 7, 2:56 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Nov 7, 3:01 PM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(language_server): handle `ServerLinter` it own restart on `workspace/didChangeConfiguration` (#15346)

> This PR refactors the configuration change handling logic by extracting the linter restart logic into a new `handle_configuration_change` method and introducing a `ToolRestartChanges` struct to encapsulate the changes.

The struct will later be used for the Formatter too.
